src/credentials/FabricTable.h

```diff
@@ -351,14 +321,53 @@ class ConstFabricIterator
 class DLL_EXPORT FabricTable
 {
 public:
+    class DLL_EXPORT FabricTableDelegate
+    {
+    public:
+        FabricTableDelegate() {}
+        virtual ~FabricTableDelegate() {}
+
+        /**
+         * Gets called when a fabric is deleted, such as on FabricTable::Delete().
+         **/
+        virtual void OnFabricDeletedFromStorage(FabricTable & fabricTable, FabricIndex fabricIndex) = 0;
+
+        /**
+         * Gets called when a fabric is loaded into Fabric Table from storage, such as
+         * during FabricTable::Init().
+         **/
+        virtual void OnFabricRetrievedFromStorage(FabricTable & fabricTable, FabricIndex fabricIndex) = 0;
+
+        /**
+         * Gets called when a fabric in Fabric Table is persisted to storage, such as
+         * on FabricTable::AddNewFabric().
+         **/
+        virtual void OnFabricPersistedToStorage(FabricTable & fabricTable, FabricIndex fabricIndex) = 0;
+
+        /**
+         * @brief If set to true, the delegate must be deleted on removal from fabric table,
+         *        since the caller relinquished ownership of a dynamically allocated object.
+         *
+         * @param deleteOnRemoval - true for a delete on removal, false if owned by caller
+         */
+        void SetMustDeleteOnRemoval(bool deleteOnRemoval) { mDeleteOnRemoval = deleteOnRemoval; }
```

Comment:
Since the rest of the functions are pure virtual, why do we have a setter implementation for this one? Why not just `virtual bool MustDeleteOnRemoval() const = 0` ?

Reply:
Because the implementation doesn't know, the caller of `FabricTable::AddFabricDelegate` knows